aws/resource_aws_acm_certificate.go

```diff
@@ -243,58 +243,70 @@ func resourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) err
-			return resource.NonRetryableError(err)
-		}
+	if err := d.Set("subject_alternative_names", cleanUpSubjectAlternativeNames(resp.Certificate)); err != nil {
+		return fmt.Errorf("Error setting subject alternative nates: %s", err)
+	}
 
-		domainValidationOptions, emailValidationOptions, err := convertValidationOptions(resp.Certificate)
+	var domainValidationOptions []map[string]interface{}
+	var emailValidationOptions []string
+	err = resource.Retry(time.Duration(1)*time.Minute, func() *resource.RetryError {
+		var err error
+		domainValidationOptions, emailValidationOptions, err = convertValidationOptions(resp.Certificate)
 
 		if err != nil {
 			return resource.RetryableError(err)
 		}
+		return nil
+	})
+	if isResourceTimeoutError(err) {
+		domainValidationOptions, emailValidationOptions, err = convertValidationOptions(resp.Certificate)
+	}
+	if err != nil {
+		return fmt.Errorf("Error getting validation options for ACM certificate: %s", err)
+	}
 
-		if err := d.Set("domain_validation_options", domainValidationOptions); err != nil {
-			return resource.NonRetryableError(err)
-		}
-		if err := d.Set("validation_emails", emailValidationOptions); err != nil {
-			return resource.NonRetryableError(err)
-		}
+	if err := d.Set("domain_validation_options", domainValidationOptions); err != nil {
+		return fmt.Errorf("Error setting domain validation options: %s", err)
+	}
+	if err := d.Set("validation_emails", emailValidationOptions); err != nil {
+		return fmt.Errorf("Error setting email validation options: %s", err)
+	}
 
-		d.Set("validation_method", resourceAwsAcmCertificateGuessValidationMethod(domainValidationOptions, emailValidationOptions))
+	d.Set("validation_method", resourceAwsAcmCertificateGuessValidationMethod(domainValidationOptions, emailValidationOptions))
 
-		if err := d.Set("options", flattenAcmCertificateOptions(resp.Certificate.Options)); err != nil {
-			return resource.NonRetryableError(fmt.Errorf("error setting certificate options: %s", err))
-		}
+	if err := d.Set("options", flattenAcmCertificateOptions(resp.Certificate.Options)); err != nil {
+		return fmt.Errorf("error setting certificate options: %s", err)
+	}
 
-		params := &acm.ListTagsForCertificateInput{
-			CertificateArn: aws.String(d.Id()),
-		}
+	input := &acm.ListTagsForCertificateInput{
+		CertificateArn: aws.String(d.Id()),
+	}
 
-		tagResp, err := acmconn.ListTagsForCertificate(params)
-		if err != nil {
-			return resource.NonRetryableError(fmt.Errorf("error listing tags for certificate (%s): %s", d.Id(), err))
-		}
-		if err := d.Set("tags", tagsToMapACM(tagResp.Tags)); err != nil {
-			return resource.NonRetryableError(err)
-		}
+	tagResp, err := acmconn.ListTagsForCertificate(input)
+	if err != nil {
+		return fmt.Errorf("error listing tags for certificate (%s): %s", d.Id(), err)
+	}
+	if err := d.Set("tags", tagsToMapACM(tagResp.Tags)); err != nil {
+		return fmt.Errorf("Error setting tags for certificate: %s", err)
```

Comment:
For Go standard library, [the preference is lowercase](https://github.com/golang/go/wiki/CodeReviewComments#error-strings):

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use `fmt.Errorf("something bad")` not `fmt.Errorf("Something bad")`, so that `log.Printf("Reading %s: %v", filename, err)` formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

We generally try to follow their practices, if possible. This condition could be enforced via `golint`, however do not currently have `golint` enabled since it raises many other linting failures we don't particularly want to go through to fix (at least today), such as initialisms. Since its not automatically enforced, its a gray area to enforce, so personally I do not require community contributors to adjust these due to review comment overload, but as maintainers we should likely _try_ to follow the preference for if/when it does become enforceable.